packages/plugin-commands-doctor/src/doctor.ts

```diff
@@ -0,0 +1,41 @@
+import renderHelp from 'render-help'
+import { docsUrl } from '@pnpm/cli-utils'
+import { logger } from '@pnpm/logger'
+import { Config } from '@pnpm/config'
+
+export const rcOptionsTypes = cliOptionsTypes
+
+export function cliOptionsTypes () {
+  return {}
+}
+
+export const shorthands = {}
+
+export const commandNames = ['doctor']
+
+export function help () {
+  return renderHelp({
+    description: 'Checks for known common issues.',
+    url: docsUrl('doctor'),
+    usages: ['pnpm doctor [options]'],
+  })
+}
+
+export async function handler (
+  opts: Pick<Config, 'failedToLoadBuiltInConfig'>
+) {
+  const { failedToLoadBuiltInConfig } = opts
+  if (
+    // If error, means loading npm builtin config failed
+    failedToLoadBuiltInConfig &&
+    // Npm installed via brew may have prefix error, related: https://github.com/pnpm/pnpm/issues/5404
+    process.platform === 'darwin' &&
+    process.env.HOMEBREW_PREFIX &&
+    process.execPath.startsWith(process.env.HOMEBREW_PREFIX)
```

Comment:
why should it only be printed on macos when installed via homebrew?

Reply:
That's a good question. I used to think this error only occurs in mac. And I rethought, as long as the builtin configuration fails to load and it contains a prefix, the error may exist. I will remove it.